src/components/ClusterManager.js

```diff
@@ -194,17 +196,19 @@ export default class ClusterManager extends PureComponent {
     return currentCluster(clusters)
   }
 
-  async executeAndRefresh(promise) {
-    try {
-      const { refreshClusters } = this.props
-      this.setState({ busy: true })
-      await promise
-      await refreshClusters()
-    } catch (error) {
-      reportError('Notebook Runtime Error', error)
-    } finally {
-      this.setState({ busy: false })
-    }
+  async executeAndRefresh(promise, waitBeforeRefreshMillis = 0) {
+    await setTimeout(async () => {
+      try {
+        const { refreshClusters } = this.props
+        this.setState({ busy: true })
+        await promise
+        await refreshClusters()
+      } catch (error) {
+        reportError('Notebook Runtime Error', error)
+      } finally {
+        this.setState({ busy: false })
+      }
+    }, waitBeforeRefreshMillis)
```

Comment:
this is needed because sometimes leo needs to do some background tasks before updating the clusters status